### ISSUE TITLE
Cannot seem to exclude Android Auto

### ISSUE THREAD

**Issue by thefunkygibbon** [zaneschepke/wgtunnel#29]
Pretty big deal for anyone who uses both wireguard and android auto in their car.   I cannot seem to find the application in the list of apps (I guess it's a system app) to exclude from the tunnel , since AA doesn't work at all if there is a VPN tunnel active. 

Running on a Samsung s23 ultra on android 13.

**Comment by zaneschepke** [comment]
Hello! This is not the first time I have heard about this issue. Android auto is a tricky situation as the initial connection is Bluetooth but then a wifi connection is generated and linked between the car and the device to increase bandwidth capabilities. This connection is not a typical wifi connection and (from my understanding) is not detected by the typical android network apis. 

As a temporary work-around, I have recently added a quick settings tile to the app which will allow you to toggle and override auto tunneling (including the VPN connection) without opening the app. This would be particularly useful in this situation. Additionally, I have added app shortcuts which can also be used or even integrated to automations like Google Routines. 

An idea was proposed that the app could have a trusted Bluetooth device name that would trigger a disablement of auto tunnel. I don't think this is a bad solution, however, I do want to consider other options before proceeding with implementing it. I am open to any other ideas you might have for a possible solution to this problem.

**Comment by thefunkygibbon** [comment]
Hi, thank you for replying. 

Yes I'd absolutely test this for you. 

Just as some further food for thought, and as i'm not a dev, I don't know for sure how they equate, but for example, Duck Duck Go's application security part of the app, which effectively creates a loopback tunnel and blocks tracking stuff.  Has, like yours does, options to say which apps installed on your device can bypass doing this.   You can select Android Auto from the list (in fact i think it might be automatically disabled by default), so, even though its not technically an 'installed app' and is part of the system these days, it still seems to be able to see it and tell it not to go through the "tunnel".

Afaik , the way that AA works is :
- Your phone and car connect via Bluetooth. This is necessary for initial pairing and for authentication.
- Your phone creates a Wi-Fi Direct connection. This is a peer-to-peer connection that allows your phone and car to communicate directly
- The phone broadcasts its Wi-Fi Direct SSID and password. This information is displayed on the car's infotainment screen.
- You connect your car to your phone's Wi-Fi Direct network. This is done using the same steps that you would use to connect to any other Wi-Fi network.
- Once your car is connected to your phone's Wi-Fi Direct network, Android Auto will launch automatically.

**Comment by thefunkygibbon** [comment]
i don't understand, sorry.  I have used the search function but i don't have anything called auto or aa or android that is obviously android auto

**Comment by thefunkygibbon** [comment]
ok i just found that the package name is apparently called com.google.android.projection.gearhead

did a search for 'gear' and it came up with android auto.   how weird.   will try it later and let you know if that works now i've excluded it

**Comment by zilexa** [comment]
![Screenshot_20230921-160937_WG_Tunnel](https://github.com/zaneschepke/wgtunnel/assets/3430004/da3ebd57-43cd-4bfc-8f12-9feb5e5d86d9)
I just search for "Android" and then it shows up in the long list when I scroll towards the bottom. In this screenshot it's the second from the top. 

Searching for "auto" didn't deliver any results nor did "Android auto". 

Android Auto Wireless will give you a red screen with a clear explanation to disable VPN otherwise it can't connect. 
I really wouldn't waste time detecting it. Because this warning will already trigger you to disable if you haven't excluded it yet.

**Comment by zaneschepke** [comment]
A big thank you to everyone who helped contribute to finding a solution for this issue! 

I will add alphabetical sorting to the packages list to assist with finding things more quickly. 

I can also look into setting Android Auto as disabled from tunneling by default.

**Comment by thefunkygibbon** [comment]
Just managed to sit down and re-read through the thread.  Apologies, I thought I was talking to Zane throughout, didn't realize that someone else chimed in. 

> Android Auto Wireless will give you a red screen with a clear explanation to disable VPN otherwise it can't connect.
> I really wouldn't waste time detecting it. Because this warning will already trigger you to disable if you haven't excluded it yet.

That isn't really the point though.  If I'm getting into my car and going for a journey, I don't want to get my phone out, unlock it, turn vpn off and invariably have to dick about a fair bit to get AA to kick in (reconnect bluetooth and suchlike)  and then remember to turn it back on again afterwards.   Hence me attempting to use tasker to do this sort of thing automatically for things which don't play nice with tunnels for apps which create a tunnel but don't give you the means to exclude certain things. 

I've still not had an opportunity to check if excluding the AA app works or not, but seeing as it does for things like Duck Duck Go and another VPN app i've used in the past, I have no reason to expect that it doesn't. 

@zaneschepke, thanks.  I'll give a quick update when i've had the chance to check. 

Although, I've just turned off my wifi and noticed that it hasn't automatically attempted to log on to the WG tunnel. despite having WG Tunnel configured to tunnel on mobile data.   i'm a bit confused now.  I used to have a persistent notification saying that it is monitoring for connectivity changes.  but that's no longer there either.
